library/std/src/primitive_docs.rs

```diff
@@ -565,8 +565,8 @@ mod prim_array {}
 ///
 /// *[See also the `std::slice` module](slice/index.html).*
 ///
-/// Slices are a view into a block of memory represented as a pointer and a
-/// length.
+/// A slice is any pointer/reference to a block of memory. They are represented
+/// as a regular pointer and a length.
```

Comment:
hmm, I am not completely sure if this rewording is an improvement here :thinking:

Reply:
In my opinion the original wording of slices being a "view", and being represented as a pointer and a length is somewhat confusing as slices, from a syntax standpoint, are themselves pointers(/references). My rewording is probably not perfect. Especially the term `regular pointer` is somewhat unspecific.

The point I wanted to make is to more closely define what exactly the term "slice" refers to. Which types fall under this definition?

The way I understand it, `&[T]`, and `&mut [T]` are definitely slices. Box<[T]> and Rc<[T]> have all the behavior of a slice, as they also Deref into a `[T]`, so I think they are slices as well. I'm not sure about `*mut [T]`/`*const [T]`. They still behave like a slice in the sense that they store a pointer + length, but you have to manually dereference them to "view" into the memory they point to, or even access the length. However with the other types, this still happens, just implicitly.

Defining a slice as "any pointer/reference" may be too broad. It's just what I think should be included. I just believe it is important to make it obvious that a slice is actually the reference, not the thing behind the reference.

I hope that explains the thought process that went into that rewording and can help to find a better one!

Reply:
Yes, this wording is definitely incorrect, but it's also tricky, because "slice" has contextual meaning. We tend to call both `[T]` and `&[T]` "slices". The issue why this is incorrect is that a slice is more than only a pointer, it's a pointer plus a length. And, as you've noted the "regular" terminology is... irregular :)

Reply:
Yeah, that both are called slice is what really had me confused when reading the docs.
Can we change that definition to define slice as only the *reference/pointer* to a `[T]` (including Box<[T]>, etc.) and use a different word for `[T]`? I feel like that would disambiguate the whole situation. The docs already call this a "sequence", so maybe that term could be used.

If that were the case, I think this wording would make the situation clear:
```
A slice is any pointer/reference to a sequence `[T]`.
They are represented as a pointer to the first element and the sequence length.
```
It would also change the "block of memory" to sequence, as at the moment, both words are used.

Reply:
After years of this, I don't think this is the kind of thing we can change, no. But it also wouldn't be something that could be done in a PR like this, it would have to be done in the reference, get the lang team involved.

Reply:
Alright, I worried that might be the case. Not sure all that trouble would be worth it, there's probably more important things to do. I can revert that part of the PR.

How do you feel about merging the paragraph about slices being twice the size of pointers to `Sized` types though? In my opinion it's good to point that out. Especially for users switching from C/C++, Rust's "fat pointers" can be unfamiliar. If you just check the docs instead of the book/reference, this is not obvious imho.